### PR TITLE
feat: use dockerized memote setup

### PR DESCRIPTION
**Pull request by mihai-sysbio**
This PR addresses [this comment](https://github.com/MetabolicAtlas/standard-GEM-validation/issues/3#issuecomment-1452610056) about using a Docker image of `memote`.
Moreover, it adds an `index.json` containing a JSON list of GEMs to be run through the validation pipeline.

### CODE REVIEW

File: .github/workflows/validation.yml [hunk 1]
```diff
@@ -19,27 +24,32 @@ jobs:
...
+        run: echo "matrix=$(cat index.json)" >> $GITHUB_OUTPUT
 
   validate:
     needs: setup
     runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/metabolicatlas/memote-docker:0.13
+      volumes:
+        - ${{ github.workspace }}:/project:rw
+      options: --user root --workdir /project
```
Comment (famosab) [inline .github/workflows/validation.yml:52]: Where would the output json for each individual model be saved and re-accessed to enable access to the MEMOTE results?

**Review by famosab — COMMENTED** [review]
I added the following question: 

> Where would the output json for each individual model be saved an re-accessed to enable access to the MEMOTE results? 

This will be relevant for a detailed report later, I think.

File: .github/workflows/validation.yml (hunk 1 above)
Comment (mihai-sysbio) [reply .github/workflows/validation.yml:52]: From the JSON, I chose to keep only the total resulting score, not the full JSON, which I thought would be too large.

**Comment by mihai-sysbio** [comment]
> This will be relevant for a detailed report later, I think.

I'm not sure who would want to see the entire report, and why. Storing large JSONs indefinitely in the results file doesn't sound like the best place, so to find another solution more details about the use-case are needed.

**Comment by mihai-sysbio** [comment]
In order to save a few Actions runs. I will go ahead and merge this PR.
Any remaining issue should be resolved in the target branch `feat/memote`. There, the discussion can continue about which JSON scores to isolate in addition to the `total` score ([original discussion](https://github.com/MetabolicAtlas/standard-GEM-validation/pull/9#discussion_r1134076183)).

**Pull request merged**